### PR TITLE
Adding Props to `NonIdealState` Component

### PR DESCRIPTION
**Pull request by marashad001**
## Changes

#1368 - Adding the ability for each node element in `NonIdealState` to have a custom `className`.

## Testing

Not there yet

## Docs

N/A

### CODE REVIEW

**Review by LostABike — APPROVED** [review]
Working as expected. LGTM 👍

File: packages/itwinui-react/src/core/NonIdealState/NonIdealState.tsx [hunk 1]
```diff
@@ -54,20 +70,69 @@ type NonIdealStateProps = {
...
-      <Box className='iui-non-ideal-state-illustration'>{svg}</Box>
-      {heading && <Box className='iui-non-ideal-state-title'>{heading}</Box>}
+      <Box
+        {...illustrationProps}
+        className={cx(
+          'iui-non-ideal-state-illustration',
+          illustrationProps?.className,
+        )}
+        ref={forwardedRef}
+        {...rest}
```
Comment (mayank99) [inline packages/itwinui-react/src/core/NonIdealState/NonIdealState.tsx:99]: `forwardedRef` and `...rest` should only be applied on the very top level element (lines 89-90), so it needs to be removed from all other elements
```suggestion
```

File: .changeset/strong-maps-explode.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@itwin/itwinui-react': major
```
Comment (mayank99) [inline .changeset/strong-maps-explode.md:2]:
```suggestion
'@itwin/itwinui-react': minor
```

please take the time to understand [semver](https://semver.org/)

File: packages/itwinui-react/src/core/NonIdealState/NonIdealState.tsx (hunk 1 above)
Comment (marashad001) [reply packages/itwinui-react/src/core/NonIdealState/NonIdealState.tsx]: fixed

**Review by mayank99 — APPROVED** [review]

**Pull request merged**